spec/unit/pdk/util_spec.rb

```diff
@@ -0,0 +1,234 @@
+require 'spec_helper'
+
+describe PDK::Util do
+  shared_context :with_version_file, version_file: true do
+    let(:version_file) { File.join('path', 'to', 'the', 'version', 'file') }
+
+    before(:each) do
+      allow(PDK::Util::Version).to receive(:version_file).and_return(version_file)
+    end
+  end
+
+  shared_context :without_version_file, version_file: false do
+    before(:each) do
+      allow(PDK::Util::Version).to receive(:version_file).and_return(nil)
+    end
+  end
+
+  describe '.find_upwards' do
+    subject { described_class.find_upwards(target, start_dir) }
+
+    let(:target) { 'metadata.json' }
+    let(:start_dir) { nil }
+    let(:actual_start_dir) { '/path/to/something/deep/in/a/module' }
+
+    before(:each) do
+      current = actual_start_dir
+      until current == '/'
+        allow(File).to receive(:directory?).with(current).and_return(true)
+        allow(File).to receive(:expand_path).with('..', current).and_return(File.dirname(current))
+        current = File.dirname(current)
+      end
+      allow(File).to receive(:directory?).with('/').and_return(true)
+      allow(File).to receive(:expand_path).with('..', '/').and_return('/')
+      allow(File).to receive(:expand_path).with(actual_start_dir).and_return(actual_start_dir)
+      allow(File).to receive(:file?).with(a_string_matching(%r{metadata\.json\Z})).and_return(false)
+    end
+
+    context 'when start_dir is nil' do
+      before(:each) do
+        allow(Dir).to receive(:pwd).and_return(actual_start_dir)
+      end
+
+      context 'and the target file exists' do
+        before(:each) do
+          allow(File).to receive(:file?).with('/path/to/something/metadata.json').and_return(true)
+        end
+
+        it { is_expected.to eq('/path/to/something/metadata.json') }
+      end
+
+      context 'and the target file does not exist' do
+        it { is_expected.to be_nil }
```

Comment:
Should this have something like

    expect(File).to receive(:file?).with('/path/to/something/metadata.json').and_return(false)

?
Maybe the code would be returning nil due to being broken, rather than due to not finding the file